lib/ExponentialBackoff.js

```diff
@@ -0,0 +1,35 @@
+'use strict';
+
+var ExponentialBackoff = function(minimum_backoff, exponent, starting_backoff) {
+    if (minimum_backoff === undefined) {
+        minimum_backoff = 50;
+    }
+
+    if (exponent === undefined) {
+        exponent = 2.0;
+    }
+
+    if (starting_backoff === undefined) {
+        starting_backoff = minimum_backoff;
+    }
+
+    var backoff = starting_backoff;
+
+    return {
+        getBackoff: function() {
```

Comment:
My stats class was meant to be a singleton, which is why I return an instance. Requiring that file multiple times will always return the same instance created the first time around. If you're not after singleton behaviour, returning the constructor is the way to go. Sorry, I didn't look through that example closely enough to catch that.

In general, because our code is `stream-centric` in import pipelines and uses `express` in API (which means things are mostly `middle-wear` functions), we don't seem to have a ton of examples of this pattern. I'm personally a fan of it, given my object-oriented background. I also like the fact that I can easily test these types of objects by monkey-patching things with mock objects where needed. Hiding things inside constructors makes it much harder to test.